react/w-books/src/services/userService.ts

```diff
@@ -0,0 +1,17 @@
+import { ApiResponse } from 'apisauce';
+import api from '../config/api';
+
+export const SIGN_UP = '';
+export const LOGIN = '/sign_in';
+
+export async function postUser(action: typeof SIGN_UP | typeof LOGIN, user:any):Promise<ApiResponse<any, any>> {
+  const response = await api.post(`users${action}`, user);
+  
+  if(response.ok){
+    const headers:any = response.headers;
+    const accessToken = headers['access-token'];
+    console.log(accessToken);
+  }
+
+  return response;
+}
```

Comment:
Create 2 functions. One for each endpoint, here they are both very similar, but normally the bodies are different and also the HTTP methods may differ. If you want to abstract the logic of the success, create a function for that